docs/notebooks/demo.ipynb

```diff
@@ -238,39 +350,64 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "grid = np.linspace(-3., 3., 20)\n",
-    "gridded = P.evaluate(grid, using='mix_mod', vb=False)\n",
-    "\n",
-    "G = qp.PDF(gridded=gridded, limits=demo_limits)\n",
-    "G.sample(100, vb=False)\n",
-    "G.plot()"
+    "# Define the x-grid locations\n",
+    "xvals = np.linspace(-5, 5, 11)\n",
+    "# Compute the corresponding y values\n",
+    "yvals = norm_dist1.pdf(xvals)\n",
+    "# Construct the PDFs using the x grid and y values\n",
+    "interp_dist = qp.interp(xvals=xvals, yvals=yvals)\n",
+    "interp_vals = interp_dist.pdf(xvals)\n",
+    "print(\"The input and output shapes are:\", xvals.shape, interp_vals.shape)\n",
+    "# Construct a single PDF for plotting\n",
+    "interp_dist1 = qp.interp(xvals=xvals, yvals=np.atleast_2d(yvals[0]))\n",
+    "fig, axes = qp.plotting.plot_native(interp_dist1, xlim=(-5., 5.), label=\"interpolated\")\n",
+    "leg = fig.legend()"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Let's unpack this a little. The `G` PDF object has an attribute `G.gridded` which contains the initial gridded function. This lookup table is used when making further approximations. To check this, let's look at whether this `G` PDF object knows what the _true_ PDF is, which approximation it's going to use, and then how it performs at making a new approximation to the PDF on a coarser grid:"
+    "## `qp` spline parameterization constructed from kernel density estimate (samples) parameterization\n",
+    "\n",
+    "This represents a set of distributions made by producing a kernel density estimate from a set of samples.\n",
+    "\n",
+    "To construct this you need to give the samples edges (shape=(npdf, Nsamples)).\n",
+    "\n",
+    "Note again that the the native visual represenation is different."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "scrolled": true
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
-    "print G.truth"
+    "# Take 100 random samples from each of 2 PDFs\n",
+    "samples = norm_dist1.rvs(size=(2, 1000))\n",
+    "# Define points at which to evaluate the kernal density estimate (KDE)\n",
+    "xvals_kde = np.linspace(-5., 5., 51)\n",
+    "# Use a utility function to construct the KDE, sample it, and they construct a spline\n",
+    "kde_dist = qp.spline_from_samples(xvals=xvals_kde, samples=samples)\n",
+    "kde_vals = kde_dist.pdf(xvals_kde)\n",
+    "print(\"The input and output shapes are:\", xvals.shape, kde_vals.shape)\n",
+    "# Construct a single PDF for plotting\n",
+    "kde_dist1 = qp.spline_from_samples(xvals=xvals_kde, samples=np.atleast_2d(samples[0]))\n",
+    "fig, axes = qp.plotting.plot_native(kde_dist1, xlim=(-5., 5.), label=\"kde\")\n",
```

Comment:
I get a type error here `only integer scalar arrays can be converted to a scalar index` that seems to be caused by scipy not getting what it's expecting in a `pdf.pdf()` call in `plotting.py`.

Reply:
It works for me.  What version of scipy are you using?